components/ntp_background_images/common/view_counter_pref_registry.cc

```diff
@@ -37,19 +40,26 @@ void RegisterProfilePrefs(user_prefs::PrefRegistrySyncable* registry) {
   registry->RegisterIntegerPref(prefs::kNewTabPageSuperReferralThemesOption,
                                 static_cast<int>(ThemesOption::kSuperReferral));
   registry->RegisterBooleanPref(prefs::kNewTabPageShowBackgroundImage, true);
-  registry->RegisterIntegerPref(prefs::kNewTabTakeoverInfobarShowCount,
-                                GetNewTabTakeoverInfobarShowCountThreshold());
+  registry->RegisterIntegerPref(prefs::kNewTabTakeoverInfobarDisplayCount,
```

Comment:
Hey - why did we decide to count down here instead of count up?

I think this will make it much harder to change the number of times we want it to show (i.e. 5 to 10 or 5 to 3).

If we recorded how often we've shown the infobar instead then we could just alter a constant:

`if (prefs_->GetInteger(kNewTabTakeoverInfobarDisplayCount) > kAllowedToShowInfobarCount)`

FWIW, that's what I naively understood the pref to be when just looking at the name (and was a bit confused when I swa that we display the infobar if `DisplayCount > 0`)

If we really want to keep it this way for some reason maybe we should update the name to `kNewTabTakeoverInfobarRemainingDisplayCount` or something and add a comment explaining how the pref is used.

Reply:
Counting down allows us not to show infobar again for users who already reached the old threshold.
So new threshold will be applied for new users only. This also simplifies the code because we don't need to introduce a new pref to track that user manually closed the infobar and we don't need to show it again.

Reply:
`kNewTabTakeoverInfobarRemainingDisplayCount` naming sounds good. I'll change it

Reply:
changed pref name to kNewTabTakeoverInfobarRemainingDisplayCount

Reply:
Hmmm, to be honest I think it still makes more sense to count up (you could potentially use a placeholder value like `11111` or `-1` to record that the user dismissed the info bar.

That said, don't want to block on on this, its just my 2c.